### DISCUSSION TITLE
Mapping in nested objects

### DISCUSSION THREAD

**Discussion by AndreHaupt85** [neuefische/questions#187] [Web] [answered]
Barbara and I are stuck in mapping over nested arrays filled with objects and other arrays.

maybe there is someone with a good hint for ur or can join us in zoom to solve the confusion in our heads that is there since the last 1,5 hours.

![Bildschirmfoto 2022-09-29 um 11 48 48](https://user-images.githubusercontent.com/109967968/193000111-936c9723-cacf-4145-b643-3e75320accca.png)

![Bildschirmfoto 2022-09-29 um 11 48 50](https://user-images.githubusercontent.com/109967968/193000125-513bc605-b469-4910-ba48-03530a4bfc77.png)

**Comment by NikRoe** [comment, marked as answer]
Hey Andre,

I would change the structure of your database data to have an array with objects where each object is built the same way. For instance with a key for name and a key for amount. That will enable you to easily map over the array.

...
```json
"spirits": [
{"name": "Vodka", "amount": "3cl"}, 
{"name": "Ouzo", "amount": "5cl"}
],
```
...

then you would be able to map over the data like this:
```js
{recipe.spirits.map((spirit, index) => {
return <li key={index}>{spirit.name}: {spirit.amount}</li>;
})}
```

**Reply by barbigrade** [reply]
¡THANK YOU SO MUCH!

**Reply by AndreHaupt85** [reply]
Barbara and me are sure u must be a wizard!

Thank u very much it worked
